### ISSUE TITLE
Create/ETL Lookup Models

### ISSUE THREAD

**Issue by NickShafferNOAA** [nwfsc-fram/boatnet#245]
Create various lookup types / models from Lookups table in OBSPROD or from paper sheets where not exists in DB, and implement with other models. 

Mostly comprehensive list (subject to change):

- [x] Beaufort
- [x] biospecimen sample method
- [x] BodyLength
- [x] Confidence
- [x] Contact type / category
- [x] catch disposition
- [x] deterrent type
- [x] discard reason
- [x] Fishery
- [x] Gear types / performance
- [x] hlfc aerial extent
- [x] hlfc horizontal dist
- [x] hlfc product delivery
- [x] interactionBehaviors
- [x] interactionOutcome
- [x] relationToObserver
- [x] species category / sub category
- [x] Trip status
- [x] Vessel  type/status
- [x] viability
- [x] Waiver type
- [x] Waiver reason
- [x] Weight methods

**Comment by NickShafferNOAA** [comment]
Generic code is now in place to easily add data (+ a view) from the "Lookups" table with only minor additions/changes to the code.

**Comment by Mak-NOAA** [comment]
@NickShafferNOAA I am working on the trips page for debriefers and have a table of all trips (database/master-dev/_design/MainDocs/_view/all-trips). Within the trips, there are a few lookup ids such as fishery, captain, debriefer, observer and I want to display the lookup value instead. Where do I get that data in couchdb? Would it be here: database/lookups-dev/_design/obs_web/_view/* OR database/master-dev/_design/LookupDocs/_view/*? 

Either way it seems like all lookups are not loaded but fisheries is in one. Can you tell me which one I should use for my code for the following lookup ids?
fishery, captain, debriefer, observer 

@sethgerou-noaa @wsmith-nwfsc FYI

**Comment by NickShafferNOAA** [comment]
@Mak-NOAA The values for captain, debriefer, and observer in master-dev don't represent anything meaningful in the database at the moment, they refer to legacy user IDs in the oracle DB. Our new "Person" model we'll be using going forward is quite different, and the translation of old records to that new model hasn't been done yet. 

As for the fishery lookup, the views-test database has a more current/accurate representation of the data than master-dev, and has the fishery lookup properly stored inside trip documents.

**Comment by Mak-NOAA** [comment]
@wsmith-nwfsc @sethgerou-noaa do we eventually expect all lookups to be in the same master-dev database? I don't want to code against another db for lookups unless that's how we are going to move forward.

**Comment by wsmith-nwfsc** [comment]
The lookups will go into master-dev and also be replicated to the readonly database lookups-dev. (Offline cliends will sync lookups-dev.) For your purposes, you can just use master-dev.

**Comment by Mak-NOAA** [comment]
Thanks

**Comment by toddhay-NOAA** [comment]
@NickShafferNOAA , is it correct that you have not integrated the lookup values into the WCGOP or A-SHOP trips and operations yet?  @Mak-NOAA is trying to show some of the data but they were still the numeric values.

**Comment by toddhay-NOAA** [comment]
Nick and I just chatted @Mak-NOAA , and most of these lookups are actually in place for WCGOP, minus those for person-related items (createdBy, captain, etc...)

**Comment by NickShafferNOAA** [comment]
I did the last couple items still sticking around on this to close this off. I was going to do maturity (already put the tiny amount of code into place for wcgop data) but then I realized at some point maturity was half-way defined as a "maturity table" in the base specimen level already, which starts to open a minor can of worms for looking at ashop maturity. 

I'm going to close this and open a ticket specifically for a common maturity lookup.
